# R Code

## Citations:

### PathFindR
Ulgen E, Ozisik O, Sezerman OU (2019). “pathfindR: An R Package for Comprehensive Identification of Enriched Pathways in Omics Data Through Active Subnetworks.” Frontiers in Genetics, 10, 858.

### Salmon
Patro, R., Duggal, G., Love, M. et al. Salmon provides fast and bias-aware quantification of transcript expression. Nat Methods 14, 417–419 (2017). https://doi.org/10.1038/nmeth.4197

### DeSEQ2
Love MI, Huber W, Anders S (2014). “Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2.” Genome Biology, 15, 550. doi: 10.1186/s13059-014-0550-8

### RADAR
Zhang, Z., Zhan, Q., Eckert, M. et al. RADAR: differential analysis of MeRIP-seq data with a random effect model. Genome Biol 20, 294 (2019) doi:10.1186/s13059-019-1915-9
